r/R/dataset-format.R

```diff
@@ -113,6 +115,35 @@ ParquetFileFormat$create <- function(...,
 #' @export
 IpcFileFormat <- R6Class("IpcFileFormat", inherit = FileFormat)
 
+#' JSON dataset file format
+#'
+#' @description
+#' A `JsonFileFormat` is a [FileFormat] subclass which holds information about how to
+#' read and parse the files included in a JSON `Dataset`.
+#'
+#' @section Factory:
+#' `JsonFileFormat$create()` can take options in the form of lists passed through as `parse_options`,
+#'  or `read_options` parameters.
```

Comment:
Is it clear to the user what they should pass in for `parse_options` and `read_options`?

Also, some of the parse options don't apply when used in a datasets context (as opposed to reading a single file).  I believe we ignore `explicit_schema` (in favor of the dataset schema) and `unexpected_field_behavior` (we default to ignore I think since we need to tolerate partial reads).

Reply:
Good point, have added a bit more in the docs about these params now.